File: 프로그래머스/1/12982. 예산/예산.py
def solution(d, budget):
    # 오름 차순으로 정렬 후 
    d.sort()
    # 순서대로 예산에서 빼면서 카운팅, 예산보다 크면 그만두기
    cnt = 0
    for need in d:
        if need <= budget:
            cnt += 1
            budget -= need
        else:
            break
    return cnt
    

File: 프로그래머스/1/12982. 예산/예산.java
import java.util.*;
class Solution {
    public int solution(int[] d, int budget) {
        /*
        정렬 후에 누적합
        buget이 누적합보다 크면 카운팅 작으면 끝
        */
        Arrays.sort(d);
        int sum = 0;
        int result = 0;
        for(int i=0; i<d.length ;i++){
            sum += d[i];
            if(sum > budget){
                return i;
            }
        }
        return d.length;
    }
}